### DISCUSSION TITLE
Google Search Plus plugin integration

### DISCUSSION THREAD

**Discussion by jjw24** [Flow-Launcher/Flow.Launcher#525] [Ideas]
I have was using the Google Search Plus plugin and thought it is quite neat that it gives you a list of your Google search directly in your results which, when selected takes you directly to the website.

I am thinking whether it would be a good idea to integrate it into the WebSearch plugin and be triggered with an action keyword similar to how you do YouTube and Github searches.

Thoughts?

You can download the plugin here:
https://github.com/jjw24/Wox.Plugin.GoogleSearch/releases/latest

**Comment by runkelstein** [comment]
I tried the Google Search Plus plugin, but for me it feels a bit too slow, I don't know why that is. But I need to wait to long to get to the results. But I like the idea of the plugin, as it can safe some time

I was myself thinking of writing a plugin which uses the "do you feel lucky" functionality from google. Because on many occasions if you find the right keywords the first result is what you are looking for. Over time you will memorize the right combinations which will get you to your goal.

E.g. if you search for "msdn console" you get instantly to the msdn documentations. Most things you need frequently can be reached by only two words.

So you just think of something and voila there is your results.

**Comment by runkelstein** [comment]
Btw. I now realize that this is not the wox project itself, but a fork of it.

I just ended up here via the github search, because I couldn't find the plugin 🤣 

So thanks for the link.

**Comment by jjw24** [comment]
> I tried the Google Search Plus plugin, but for me it feels a bit too slow, I don't know why that is. But I need to wait to long to get to the results. But I like the idea of the plugin, as it can safe some time
> 
> I was myself thinking of writing a plugin which uses the "do you feel lucky" functionality from google. Because on many occasions if you find the right keywords the first result is what you are looking for. Over time you will memorize the right combinations which will get you to your goal.
> 
> E.g. if you search for "msdn console" you get instantly to the msdn documentations. Most things you need frequently can be reached by only two words.
> 
> So you just think of something and voila there is your results.

Yeah I agree, sometimes I just need the first or one of top 6 results so it I thought it is a great idea.

The do you feel lucky Google search gives random results though right, preferably always the same result is better.

**Comment by runkelstein** [comment]
Is it really random? I was thinking it is the first result, but if it is random then it is not realy useful.

But nonetheless, parsing the general search and selecting the first result should not be hard either.

I just looked at the search plugin of wox, and realized there is a long list of keywords.

One could generalize the idea and just add a letter to the existing keywords, like 
gf (google first)
wikif (wiki first) 

etc.

**Comment by runkelstein** [comment]
https://www.google.com/search?btnI=1&q=msdn%20console

I always get to the same result. There is just a page in between which I would need to skip.

**Comment by runkelstein** [comment]
The Google Search Plus plugin almost works like the lucky button, but it just takes to long loading up the results. On my machine a few seconds.

But also here a combination of the "lucky" functionality and the search is thinkable. Like if you hit enter immediately before the first result comes up, the first result will open in your browser.

Meanwhile the entered search text could apear in the first entry, like it happens with the default search plugin, but get replaced once the search results are visible.

I just want something as quick as opening up a file.

**Comment by jjw24** [comment]
> But nonetheless, parsing the general search and selecting the first result should not be hard either.
> 
> I just looked at the search plugin of wox, and realized there is a long list of keywords.
> 
> One could generalize the idea and just add a letter to the existing keywords, like 
> gf (google first)
> wikif (wiki first) 
> 
> etc.

Directly to the first result using action keyword is a good idea 👍

**Comment by jjw24** [comment]
> The Google Search Plus plugin almost works like the lucky button, but it just takes to long loading up the results. On my machine a few seconds.

Yep on my machine takes a second or two as well. Something worth look out for during implementation.

**Comment by pota2dev** [comment]
Can we expect any rework on it?

**Reply by jjw24** [reply]
It has been pulled from the Plugin Store because it is no longer working and I am not able to dedicate time to fix it https://github.com/jjw24/Wox.Plugin.GoogleSearch#temporarily-removed-from-plugin-store.

The project is still open for someone to continue its maintenance either by forking or creating a PR to fix the bug (which I am more than happy to merge in and re-release to the store).
